src/coreclr/nativeaot/System.Private.CoreLib/src/System/ComAwareWeakReference.NativeAot.cs

```diff
@@ -2,32 +2,55 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Runtime.CompilerServices;
 
 #if FEATURE_COMINTEROP || FEATURE_COMWRAPPERS
 
-#pragma warning disable IDE0060
-
 namespace System
 {
     internal sealed partial class ComAwareWeakReference
     {
-        internal static object? ComWeakRefToObject(IntPtr pComWeakRef, long wrapperId)
+        // We don't want to consult ComWrappers if no RCW objects have been created.
+        // In addition we don't want a direct reference to ComWrappers to allow for better
+        // trimming.  So we instead make use of delegates that ComWrappers registers when
+        // it is used.
+        private static unsafe delegate*<IntPtr, long, object?> ComWeakRefToObjectCallback;
+        private static unsafe delegate*<object, bool> PossiblyComObjectCallback;
+        private static unsafe delegate*<object, long*, IntPtr> ObjectToComWeakRefCallback;
+
+        internal static unsafe void InitializeCallbacks(
+            delegate*<IntPtr, long, object?> comWeakRefToObject,
+            delegate*<object, bool> possiblyComObject,
+            delegate*<object, long*, IntPtr> objectToComWeakRef)
+        {
+            // PossiblyComObjectCallback is initialized last to avoid any potential races
+            // where functions are being called while initialization is happening.
+            ComWeakRefToObjectCallback = comWeakRefToObject;
+            ObjectToComWeakRefCallback = objectToComWeakRef;
+            PossiblyComObjectCallback = possiblyComObject;
+        }
+
+        internal static unsafe object? ComWeakRefToObject(IntPtr pComWeakRef, long wrapperId)
         {
-            // NativeAOT support for COM WeakReference is NYI
-            throw new NotImplementedException();
+            return ComWeakRefToObjectCallback != null ? ComWeakRefToObjectCallback(pComWeakRef, wrapperId) : null;
         }
 
-        internal static bool PossiblyComObject(object target)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Nit: is this still needed and/or doing anything now that the method is only ever called via fnptr?

Reply:
Well this is the one calling the function pointer, so it can still be useful I think.  Let me know if otherwise.  But it probably isn't useful in the actual function in the other file.

Reply:
Oops sorry, added a comment on the wrong line, I meant to add this one to this method:

https://github.com/dotnet/runtime/blob/032b7d35e3a6221b7fc9cc9d1f5cfba2baad0d55/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs#L1566-L1570

Reply:
Addressed.